vim_onenote.ahk

```diff
@@ -45,65 +49,160 @@ IsLastKey(key)
     return (A_PriorHotkey == key and A_TimeSincePriorHotkey < 400)
 }
 
-; ESC enters Normal Mode
-ESC::
-	Suspend, Off
-    ; send escape on through to exit from find
-    SendInput, {ESC} 
-	ToolTip, OneNote Vim Command Mode Active, 0, 0
-return
+
 
 ;--------------------------------------------------------------------------------
 ; Return to InsertMode
 InsertMode:
     ToolTip
     Suspend, On
+    hotkey, ESC, on
 return
 
 ;--------------------------------------------------------------------------------
+
+;  ctrl + [, ESC enter Normal Mode.
+; Not using ctrl + c, as people may still want to use for copying.
+;^c::
+^[::
+ESC::
+    ; Is not affected by insertmode's suspend
+    suspend, permit
+    gosub NormalMode
+return
+
+; imap workings (eg jj) currently can't be implemented because of how 
+; insert mode works.
+
+
+NormalMode:
+    Suspend, Off
+    ToolTip, OneNote Vim Command Mode Active, 0, 0
+return
+
+
+;--------------------------------------------------------------------------------
 +i::
-    SendInput, {Home}
+    send {Home}
     Gosub InsertMode
 return 
 
 i::Gosub InsertMode
 return 
 
 ;--------------------------------------------------------------------------------
+PrepareClipboard(){
+    ; push clipboard to local variable
+    ClipSaved := ClipboardAll
+    ; Clear clipboard to avoid errors
+    Clipboard :=
+}
+
+Copy(){
```

Comment:
Could you explain what you mean by 
>copy doing copy and save
?
The reason paste and restore are separate is that they aren't always done together.